components/list/list-item-drag-mixin.js

```diff
@@ -0,0 +1,115 @@
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { html } from 'lit-element/lit-element.js';
+import { nothing } from 'lit-html';
+
+export const ListItemDragMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			draggable: { type: Boolean, reflect: true },
+			key: { type: String, reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this._itemDragId = getUniqueId();
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (!this.key) {
+			this.draggable = false;
+		}
+	}
+
+	_dispatchDragEvent(detail) {
+		this.dispatchEvent(new CustomEvent('d2l-list-item-position', {
+			detail: detail,
+			bubbles: true
+		}));
+		return true;
+	}
+
+	keyboardMode(isEnabled) {
+		if (isEnabled) {
+			this.dispatchEvent(new CustomEvent('d2l-list-item-drag-keyboard-mode', {
+				detail: null,
+				bubbles: true
+			}));
+		}
+	}
+
+	_renderDragHandle() {
+		return this.draggable ? html`
+			<div
+				id="${this._itemDragId}"
+				class="d2l-list-item-drag"
+				@drop="${this._dropHandlers}"
+				@dragover="${this._dragOverHandlers}"
```

Comment:
These two events won't be connected here and can be removed for now. Future story will cover all the drop and drag over stuff. It will need to spawn something over the entire object the action slot.